Back from Settings -> App crash

Click on Settings in Sidemenu and follow with back in nav bar or with hardware back key crashs app

Exception: Receiver not registered

/AndroidRuntime: FATAL EXCEPTION: main
                                                   Process: de.qabel.qabel, PID: 21185
                                                   **java.lang.RuntimeException: Unable to destroy activity {de.qabel.qabel/de.qabel.qabelbox.settings.SettingsActivity}: java.lang.IllegalArgumentException: Receiver not registered: de.qabel.qabelbox.settings.fragments.SettingsFragmen**t$1@16ee816f
                                                       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3916)
                                                       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3934)
                                                       at android.app.ActivityThread.access$1400(ActivityThread.java:151)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1408)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:135)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5499)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at java.lang.reflect.Method.invoke(Method.java:372)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:737)
                                                    **Caused by: java.lang.IllegalArgumentException: Receiver not registered: de.qabel.qabelbox.settings.fragments.SettingsFragmen**t$1@16ee816f
                                                       at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:783)
                                                       at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1666)
                                                       at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:511)
                                                       at de.qabel.qabelbox.settings.fragments.SettingsFragment.onDetach(SettingsFragment.java:85)
                                                       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1021)
                                                       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
                                                       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1049)
                                                       at android.app.FragmentManagerImpl.dispatchDestroy(FragmentManager.java:1897)
                                                       at android.app.Activity.performDestroy(Activity.java:6168)
                                                       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1157)
                                                       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3896)
                                                       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3934) 
                                                       at android.app.ActivityThread.access$1400(ActivityThread.java:151) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1408) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:135) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5499) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at java.lang.reflect.Method.invoke(Method.java:372) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:737) 

--- a/qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.kt
+++ b/qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.kt
@@ -41,7 +41,10 @@ import de.qabel.qabelbox.dagger.modules.MainActivityModule
 import de.qabel.qabelbox.fragments.BaseFragment
 import de.qabel.qabelbox.fragments.IdentitiesFragment
 import de.qabel.qabelbox.fragments.QRcodeFragment
-import de.qabel.qabelbox.helper.*
+import de.qabel.qabelbox.helper.AccountHelper
+import de.qabel.qabelbox.helper.CacheFileHelper
+import de.qabel.qabelbox.helper.Sanity
+import de.qabel.qabelbox.helper.UIHelper
 import de.qabel.qabelbox.navigation.MainNavigator
 import de.qabel.qabelbox.settings.SettingsActivity
 import de.qabel.qabelbox.sync.FirebaseTopicManager
@@ -408,6 +411,7 @@ class MainActivity : CrashReportingActivity(),
             withIdentifier(R.id.nav_settings.toLong())
             withName(R.string.action_settings)
             withIcon(R.drawable.settings)
+            withSelectable(false)
         }
         val tellAFriend = SecondaryDrawerItem().apply {
             withIdentifier(R.id.nav_tellafriend.toLong())

--- a/qabelbox/src/main/java/de/qabel/qabelbox/settings/fragments/SettingsFragment.java
+++ b/qabelbox/src/main/java/de/qabel/qabelbox/settings/fragments/SettingsFragment.java
@@ -87,16 +87,16 @@ public class SettingsFragment extends PreferenceFragment {
     }
 
     @Override
-    public void onAttach(Context context) {
-        super.onAttach(context);
+    public void onResume() {
+        super.onResume();
         getActivity().registerReceiver(accountBroadcastReceiver,
                 new IntentFilter(QblBroadcastConstants.Account.ACCOUNT_CHANGED));
     }
 
     @Override
-    public void onDetach() {
+    public void onPause() {
+        super.onPause();
         getActivity().unregisterReceiver(accountBroadcastReceiver);
-        super.onDetach();
     }
 
     private void refreshAccountData() {

